--- server.c
+++ server.c	2000/04/18 08:02:51
@@ -826,6 +826,7 @@
 		yp_msg ("ypproc_xfr: refuse to transfer %s from %s, not master",
 			xfr->map_parms.map, inet_ntoa (rqhost->sin_addr));
 
+	      ypdb_close (dbp);
 	      result.xfrstat = YPXFR_NODOM;
 	      return &result;
 	    }
@@ -837,6 +838,7 @@
 	  if (debug_flag)
 	    yp_msg ("\t->Ignored (no YP_MASTER_NAME key in local map)\n");
 
+	  ypdb_close (dbp);
 	  result.xfrstat = YPXFR_REFUSED;
 	  return &result;
 	}
@@ -1241,7 +1243,8 @@
 	      if(gethostname (hostbuf, sizeof (hostbuf) - 1) != 0)
 		{
 		  perror ("gethostname");
-		  exit (1);
+		  ypdb_close (dbp);
+		  return NULL;
 		}
 #if USE_FQDN
 	      else
@@ -1254,8 +1257,8 @@
 		    {
 		      strncpy (hostbuf, hp->h_name, sizeof (hostbuf) - 1);
 		      hostbuf[sizeof (hostbuf) - 1] = '\0';
-	}
-    }
+		    }
+		}
 #endif
 
 	      if ((result.peer = strdup (hostbuf)) == NULL)
